tests/testthat/test-update_snapshot.R

```diff
@@ -85,37 +76,177 @@ test_that("update_snapshot() works", {
     if (inherits(conn, "SQLiteConnection")) {
       types <- types |>
         purrr::map_if(~ identical(., "POSIXct"), "character") |> # SQLite does not support POSIXct
-        purrr::map_if(~ identical(., "logical"), "numneric") |>  # SQLite does not support logical
+        purrr::map_if(~ identical(., "logical"), "numeric") |>  # SQLite does not support logical
         as.character()
     }
 
     checkmate::expect_data_frame(logs, nrows = 1, types)
 
+    # Check the content of the log table
+    expect_identical(as.character(logs$date), as.character(timestamp))
+
+    db_table_id <- id(db_table, conn)
+    if ("catalog" %in% colnames(logs)) expect_identical(logs$catalog, purrr::pluck(db_table_id, "name", "catalog"))
+    expect_identical(logs$schema, purrr::pluck(db_table_id, "name", "schema"))
+    expect_identical(logs$table, purrr::pluck(db_table_id, "name", "table"))
+
+    expect_identical(logs$n_insertions, 32L)
+    expect_identical(logs$n_deactivations, 0L)
+    expect_true(as.logical(logs$success))
+    expect_identical(logs$message, NA_character_)
+
+
+    # Clean up the logs
+    unlink(logger$log_realpath)
+
+    close_connection(conn)
+  }
+})
+
+
+test_that("update_snapshot() can add a new snapshot", {
+  for (conn in get_test_conns()) {
+
+    # Modify snapshot and run update step
+    .data <- mtcars |>
+      dplyr::mutate(hp = dplyr::if_else(hp > 130, hp - 10, hp)) |>
+      dplyr::copy_to(conn, df = _, name = unique_table_name())
+
+    # Configure the logger for this update
+    db_table <- "test.SCDB_tmp1"
+    timestamp <- "2022-10-03 09:00:00"
+
+    logger <- Logger$new(
+      db_table = db_table,
+      timestamp = timestamp,
+      log_path = NULL,
+      log_table_id = "test.SCDB_logs",
+      log_conn = conn,
+      output_to_console = FALSE
+    )
+
+
+    # Update
+    # This is a simple update where 15 rows are replaced with 15 new ones on the given date
+    update_snapshot(.data, conn, db_table, timestamp, logger = logger)
+
+    # Check the snapshot has updated correctly
+    target <- dplyr::tbl(conn, id(db_table, conn))
+    expect_identical(
+      slice_time(target, "2022-10-01 09:00:00") |>
+        dplyr::select(!c("from_ts", "until_ts", "checksum")) |>
+        dplyr::collect() |>
+        dplyr::arrange(wt, qsec),
+      mtcars |>
+        dplyr::arrange(wt, qsec) |>
+        tibble::as_tibble()
+    )
+    expect_equal(
+      nrow(slice_time(target, "2022-10-01 09:00:00")),
+      nrow(mtcars)
+    )
+
+    expect_identical(
+      slice_time(target, "2022-10-03 09:00:00") |>
+        dplyr::select(!c("from_ts", "until_ts", "checksum")) |>
+        dplyr::collect() |>
+        dplyr::arrange(wt, qsec),
+      .data |>
```

Comment:
A little confusing that you change .data instead of creating .data_modified

Reply:
I am not changing the data only the order. The order of the records is not (always) respected when using databases, so to compare that the data are equal, I need to sort them first to create the same order for both.